Python/Source/Lift.cpp

```diff
@@ -0,0 +1,90 @@
+#include <vector>
+#include "pybind11/functional.h"
+#include "Autobox.hpp"
+#include "Rover/Box.hpp"
+#include "Rover/Evaluator.hpp"
+
+using namespace pybind11;
+
+namespace {
+  using namespace Rover;
+  using ArgsCont = std::vector<Box<object>>;
+  using KwargsCont = std::vector<std::pair<handle,
+    Box<object>>>;
+  
+  ArgsCont args_apply(const std::function<Box<object>(object&&)>& func,
+      tuple&& args) {
+    auto result = ArgsCont();
+    for(auto i = 0U; i < args.size(); ++i) {
+      result.push_back(func(std::move(args[i])));
+    }
+    return result;
+  }
+
+  KwargsCont args_apply(const std::function<Box<object>(object&&)>& func,
+      dict&& args) {
+    auto result = KwargsCont();
+    for(auto& arg : args) {
+      auto value = func(reinterpret_borrow<object>(std::move(arg.second)));
+      result.emplace_back(std::move(arg.first), std::move(value));
+    }
+    return result;
+  }
+
+  tuple args_apply(const std::function<object(Box<object>&)>& func,
+      ArgsCont& cont) {
```

Comment:
I cannot, because ArgsCont contains Box<object>'s, and the function uses their generate methods, which are in general non-const (as in all our cases besides Constant).

Reply:
Yep, ignore this comment then.